pkg/apitypes/managed_tx.go

```diff
@@ -179,6 +181,85 @@ func (mtx *ManagedTX) SetSequence(i int64) {
 	mtx.SequenceID = fmt.Sprintf("%.12d", i)
 }
 
+// ApplyExternalTxUpdates applies the external updates to the managed transaction
+// and returns the updates that were applied
+// ApplyExternalTxUpdates applies the external updates to the managed transaction
+// and returns the updates that were applied, whether any update occurred, and a map
+// of changed fields with their old and new values for easy JSON marshalling.
+func (mtx *ManagedTX) ApplyExternalTxUpdates(txUpdate TXUpdatesExternal) (txUpdates TXUpdates, updated bool, valueChangeMap map[string]map[string]string) {
+	txUpdates = TXUpdates{}
+	valueChangeMap = make(map[string]map[string]string)
+
+	if txUpdate.To != nil && mtx.To != *txUpdate.To {
+		valueChangeMap["to"] = map[string]string{
```

Comment:
The `old`/`new` item needs to be an API structured type, rather than string fields please.

I haven't reached the point in understanding what this is used for:

> ... and a map
> // of changed fields with their old and new values for easy JSON marshalling.